src/transformers/models/deberta_v2/modeling_tf_deberta_v2.py

```diff
@@ -138,9 +115,26 @@ def __init__(self, drop_prob, **kwargs):
         super().__init__(**kwargs)
         self.drop_prob = tf.convert_to_tensor(drop_prob, dtype=tf.float32)
 
+    @tf.custom_gradient
+    def xdropout(self, input):
+        """
+        Applies dropout to the input, as vanilla dropout, but also scales the remaining elements up by 1/drop_prob.
+        """
+        mask = tf.cast(
+            1 - tf.compat.v1.distributions.Bernoulli(probs=1 - self.drop_prob).sample(sample_shape=shape_list(input)),
+            tf.bool,
+        )
+        scale = tf.convert_to_tensor(1.0 / (1 - self.drop_prob), dtype=tf.float32)
+        input = tf.cond(self.drop_prob > 0, lambda: tf.where(mask, 0.0, input) * scale, lambda: input)
+
+        def grad(upstream):
+            return tf.cond(scale > 1, lambda: tf.where(mask, 0.0, upstream) * scale, lambda: upstream)
```

Comment:
Is there a reason to use `tf.cond` over standard Python `if` statements here? It feels harder to read, and `self.drop_prob` and `scale` shouldn't vary between calls!

Reply:
The original code converted `self.drop_prob` into a tensor, so all if/elses had to be `tf.cond`. 

But there is no good reason for `self.drop_prob` to be a tensor, so I've removed the conversion and replaced `tf.cond` by if/elses! Thanks for the suggestion 👍